src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -129,9 +135,20 @@ public ReadOnlyBytes Slice(int index)
             else {
                 var first = First;
                 if (first.Length > index) {
-                    return new ReadOnlyBytes(first.Slice(index), _rest);
+                    var slice = first.Slice(index);
+                    if(slice.Length > 0 || _rest!=null) {
+                        return new ReadOnlyBytes(slice, _rest);
```

Comment:
I see. To confirm, this should be an 'or' (as in we could end up with ReadOnlyBytes (0, _rest) or ReadOnlyBytes(slice, null)? Both are valid?
```C#
if(slice.Length > 0 || _rest!=null) { 
```